src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -619,17 +634,77 @@ ServerHandlerImp::processRequest (Port const& port,
 // Returns `true` if the HTTP request is a Websockets Upgrade
 // http://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header#Use_with_WebSockets
 bool
-ServerHandlerImp::isWebsocketUpgrade (http_request_type const& request)
+ServerHandlerImp::isWebsocketUpgrade(
+    http_request_type const& request) const
 {
     if (is_upgrade(request))
-        return request.headers["Upgrade"] == "websocket";
+        return beast::detail::ci_equal(
+            request.headers["Upgrade"], "websocket");
     return false;
 }
 
+bool
+ServerHandlerImp::isStatusRequest(
+    http_request_type const& request) const
+{
+    return request.version >= 11 && request.url == "/" &&
+        request.body.size() == 0 && request.method == "GET";
+}
+
+Handoff
+ServerHandlerImp::statusRequest(
+    http_request_type const& request) const
+{
+    using namespace beast::http;
+    Handoff handoff;
+    response_v1<string_body> msg;
+    std::string reason;
+    if (app_.serverOkay(reason))
```

Comment:
This needs a comment at least explaining that its for load balancing and reporting when the server is overloaded